Amplify/Categories/DataStore/Model/Model.swift

```diff
@@ -30,3 +30,16 @@ public protocol Model: Codable {
     var id: Identifier { get }
 
 }
+
+public protocol JsonModel: Model {
+
+    /// Return the value for key
+    func internal_value(for key: String) -> Any?
```

Comment:
1. Whence the underscore? This shouldn't even pass linting
1. I'm not really sure what an "internal value" means. What is this actually doing?
1. Follow Swift naming conventions for the protocol: `JSONModel`, but what about this protocol is specific to the JSON format? What is it really doing?